# Exercício 3– Teste a hipótese do salário ser igual a 12. Qual sua conclusão?

In [1]:
# importa biblioteca pandas
import pandas as pd


# importa biblioteca scipy
import scipy.stats as stats


In [2]:
# local do arquivo
Caminho='/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'


In [3]:
#Traz o arquivo para o Python
df=pd.read_excel(Caminho)

In [4]:
df.head( )

ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out

In [5]:
# trazer apenas a variável idade para executar o teste
base_idade=df['idade']


In [6]:
# executa o teste t Testando H0: idade= 32 e H1:idade ≠ 32
stats.ttest_1samp(base_idade, 32)

TtestResult(statistic=8.458459436515964, pvalue=4.4022567945723766e-16, df=425)

In [7]:
p_value = 4.4022567945723766e-16
alpha = 0.05

if p_value < alpha:
    print("Rejeitamos a hipótese nula (H0).")
else:
    print("Não rejeitamos a hipótese nula (H0).")

Rejeitamos a hipótese nula (H0).


In [8]:
#Lembrando a média
df.idade.mean()

34.62206572769953

In [9]:
# executa o teste t Testando H0: idade= 34 e H1:idade ≠ 34
stats.ttest_1samp(base_idade, 34)
# PARA 95% DE CONFIANÇA REJEITO H0 SE PVALUE < 0.05

TtestResult(statistic=2.0067070283587505, pvalue=0.045413783138037615, df=425)

# Adicional

In [10]:
# Definir o nível de significância
alpha = 0.05

# Valor-p do teste t
p_value = 0.045413783138037615

# Decidir se rejeitamos ou não a hipótese nula
if p_value < alpha:
    print("Rejeitamos a hipótese nula (H0).")
else:
    print("Não rejeitamos a hipótese nula (H0).")

Rejeitamos a hipótese nula (H0).


## Com um valor-p de 0.045, que é menor que o nível de significância de 0.05, rejeitamos a hipótese nula. Isso indica que há evidências estatisticamente significativas para sugerir que a idade média dos funcionários é diferente de 34.

In [11]:
# executa o teste t Testando H0: idade= 35 e H1:idade ≠ 35
stats.ttest_1samp(base_idade, 35)
# PARA 95% DE CONFIANÇA REJEITO H0 SE PVALUE < 0.05

TtestResult(statistic=-1.2191691757198562, pvalue=0.22345620342178218, df=425)

In [12]:
# Definir o nível de significância
alpha = 0.05

# Valor-p do teste t
p_value = 0.22345620342178218

# Decidir se rejeitamos ou não a hipótese nula
if p_value < alpha:
    print("Rejeitamos a hipótese nula (H0).")
else:
    print("Não rejeitamos a hipótese nula (H0).")

Não rejeitamos a hipótese nula (H0).


### Com um valor-p de 0.2234, que é maior que o nível de significância de 0.05, não rejeitamos a hipótese nula. Isso significa que não há evidências suficientes para concluir que a idade média dos funcionários é diferente do valor hipotetizado (32, 35 ou qualquer outro valor que estava sendo testado)

### Repondendo a questao 3

### Teste a hipótese do salário ser igual a 12. Qual sua conclusão?

In [13]:
import pandas as pd
import scipy.stats as stats


In [14]:
# Local do arquivo
Caminho = '/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'

In [15]:
# Carregar os dados
df = pd.read_excel(Caminho)

In [16]:
#Traz o arquivo para o Python
df=pd.read_excel(Caminho)

In [17]:
df.head( )

ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out

In [18]:
# Selecionar a variável de interesse (salário)
salarios = df['salario']

In [19]:
# Executar o teste t Testando H0: salario = 12 e H1: salario ≠ 12
resultado_teste = stats.ttest_1samp(salarios, 12)
resultado_teste

TtestResult(statistic=-4.500727991746298, pvalue=8.755117588192511e-06, df=425)

In [20]:
# Obter o valor-p do resultado do teste
p_value = resultado_teste.pvalue
p_value

8.755117588192511e-06

In [21]:
# Definir o nível de significância
alpha = 0.05

In [22]:
# Interpretar o resultado
p_value = 8.755117588192511e-06
alpha = 0.05

if p_value < alpha:
    print("Rejeitamos a hipótese nula (H0).")
else:
    print("Não rejeitamos a hipótese nula (H0).")

Rejeitamos a hipótese nula (H0).


In [23]:
# Definir a conclusão com base no valor-p e no nível de significância
if p_value < alpha:
    conclusao = "Rejeitamos a hipótese nula (H0)."
else:
    conclusao = "Não rejeitamos a hipótese nula (H0)."

# Exibir o resultado do teste e a conclusão
print(f"Estatística do teste t: {resultado_teste.statistic}")
print(f"Valor-p: {resultado_teste.pvalue}")
print(conclusao)

Estatística do teste t: -4.500727991746298
Valor-p: 8.755117588192511e-06
Rejeitamos a hipótese nula (H0).


### Análise e Conclusão com bano valor-p e no nível de significância

No teste de hipótese realizado, testamos a hipótese nula (H0) de que o salário médio dos funcionários é igual a 12 contra a hipótese alternativa (H1) de que o salário médio dos funcionários é diferente de 12.

Os resultados do teste t foram os seguintes:
- Estatística do teste t: -4.500727991746298
- Valor-p: 8.755117588192511e-06

Com um valor-p de aproximadamente 8.76e-06, que é significativamente menor que o nível de significância de 0.05, rejeitamos a hipótese nula (H0). Isso indica que há evidências estatisticamente significativas para sugerir que o salário médio dos funcionários é diferente de 12.

Portanto, a conclusão do teste é que rejeitamos a hipótese nula (H0) e aceitamos a hipótese alternativa (H1), indicando que o salário médio dos funcionários não é igual a 12.